runbot/controllers/frontend.py

```diff
@@ -704,16 +706,48 @@ def dockerfile_content(self, dockerfile_id=None, version=None, docker_tag=None,
             dockerfile = dockerfile_sudo.search([('image_tag', '=', docker_tag)])
         elif version:
             dockerfile = dockerfile_sudo.search([('version_ids.name', '=', version)])
+        elif dockerfile_result_id:
+            dockerfile_result = request.env['runbot.docker_build_result'].browse(int(dockerfile_result_id)).exists()
+            dockerfile = dockerfile_result.dockerfile_id
         else:
             raise NotFound
 
         if dockerfile.public_visibility:
-            return Response(response=dockerfile.layer_ids.render_layers({
+            if dockerfile_result:
+                render = dockerfile_result.content
+            else:
+                render = dockerfile.layer_ids.render_layers({
                 'USERUID': '${USERUID}',
                 'USERGID': '${USERGID}',
                 'USERNAME': '${USERNAME}',
-            }), status=200, mimetype='text/plain')
+            })
+            return Response(response=render, status=200, mimetype='text/plain')
 
         if dockerfile:
             _logger.error('Trying to access a non public docker image')
         raise NotFound
+
+    @route([
+        '/runbot/dockerfile_result/<int:dockerfile_result_id>',
+    ], type='http', auth='public', sitemap=False)
+    def dockerfile_result(self, dockerfile_result_id=None, **kwargs):
+        dockerfile_result_sudo = request.env['runbot.docker_build_result'].sudo()
+        if dockerfile_result_id:
+            dockerfile_result = dockerfile_result_sudo.browse(dockerfile_result_id).exists()
+        else:
+            raise NotFound
+
+        if not dockerfile_result:
+            raise NotFound
+
+        dockerfile = dockerfile_result.dockerfile_id
+
+        if not dockerfile_result.dockerfile_id.public_visibility:
```

Comment:
Could be good to allow admins to view any dockerfile result:
```suggestion
        if not (dockerfile.public_visibility or request.env.user.has_group('runbot.group_runbot_admin')):
```